### PR TITLE
OADP-774 must-gather: add timeout to velero logs/describe, var typos, remove duplicate logs, add `make run`

### PR DESCRIPTION
**Pull request by kaovilai**
[OADP-774](https://issues.redhat.com//browse/OADP-774)

Signed-off-by: Tiger Kaovilai <tkaovila@redhat.com>

### CODE REVIEW

**Review by savitharaghunathan — APPROVED** [review]
/lgtm

File: must-gather/collection-scripts/logs/gather_logs_pvr [hunk 1]
```diff
@@ -9,12 +9,8 @@ max_parallelism=$4
 pvr=$5
 object_collection_path=$6
 
-# Gather PVR describe and logs
+# Gather PVR describe
 mkdir -p ${object_collection_path}
 oc describe podvolumerestores.velero.io ${pvr} --namespace ${ns} &> "${object_collection_path}/pvr-describe-${pvr}.txt"
-echo "[cluster=${cluster}][ns=${ns}][pod=${pod}] Collecting Pod logs..."
-oc logs --all-containers --namespace ${ns} ${pod} --since ${logs_since} &> "${object_collection_path}/current.log" &
-echo "[cluster=${cluster}][ns=${ns}][pod=${pod}] Collecting previous Pod logs..."
```
Comment (kaovilai) [inline must-gather/collection-scripts/logs/gather_logs_pvr:17]: This is where the empty `[pod=]` logs came from

**Comment by kaovilai** [comment]
everything seems to work

File: must-gather/collection-scripts/logs/gather_logs_restore [hunk 2]
```diff
@@ -12,14 +12,14 @@ timeout=$7
...
+    oc -n ${ns} exec --request-timeout=${timeout} $(oc -n ${ns} get po -l component=velero -o custom-columns=name:.metadata.name --no-headers) -- /bin/bash -c "timeout 30s /velero describe restore ${restore} --insecure-skip-tls-verify=${skip_tls} --details" &> "${object_collection_path}/restore-describe-${restore}.txt" &
 fi
 echo "[cluster=${cluster}][ns=${ns}] Gathering 'velero restore logs ${restore}'"
-oc -n ${ns} exec $(oc -n ${ns} get po -l component=velero -o custom-columns=name:.metadata.name --no-headers) -- /bin/bash -c "/velero restore logs ${restore} --insecure-skip-tls-verify=${skip_tls} --timeout=30s" &> "${object_collection_path}/restore-${restore}.log" &
+oc -n ${ns} exec $(oc -n ${ns} get po -l component=velero -o custom-columns=name:.metadata.name --no-headers) -- /bin/bash -c "timeout 30s /velero restore logs ${restore} --insecure-skip-tls-verify=${skip_tls} --timeout=30s" &> "${object_collection_path}/restore-${restore}.log" &
```
Comment (kaovilai) [inline must-gather/collection-scripts/logs/gather_logs_restore:23]: Here we are forcing velero CLI commands that involve downloadrequest.Stream to timeout which will resolve issues related to must-gather getting stuck when querying from nonexistent backup storage location.

Comment (kaovilai) [reply must-gather/collection-scripts/logs/gather_logs_restore:23]: avoids https://github.com/vmware-tanzu/velero/issues/5324

File: must-gather/collection-scripts/logs/gather_logs_backup [hunk 3]
```diff
@@ -15,11 +15,11 @@ skip_tls=$8
...
+    oc -n ${ns} exec $(oc -n ${ns} get po -l component=velero -o custom-columns=name:.metadata.name --no-headers) -- /bin/bash -c "timeout 30s /velero describe backup ${backup} --insecure-skip-tls-verify=${skip_tls} --details" &> "${object_collection_path}/backup-describe-${backup}.txt" &
 else
-    oc -n ${ns} exec --request-timeout=${timeout} $(oc -n ${ns} get po -l component=velero -o custom-columns=name:.metadata.name --no-headers) -- /bin/bash -c "/velero describe backup ${backup} --insecure-skip-tls-verify=${skip_tls} --details" &> "${object_collection_path}/backup-describe-${backup}.txt" &
+    oc -n ${ns} exec --request-timeout=${timeout} $(oc -n ${ns} get po -l component=velero -o custom-columns=name:.metadata.name --no-headers) -- /bin/bash -c "timeout ${timeout} /velero describe backup ${backup} --insecure-skip-tls-verify=${skip_tls} --details" &> "${object_collection_path}/backup-describe-${backup}.txt" &
```
Comment (savitharaghunathan) [inline must-gather/collection-scripts/logs/gather_logs_backup:20]: do we need `--request-timeout=${timeout}` here if timeout is getting passed in the velero container?

**Comment by savitharaghunathan** [comment]
@kaovilai should there be a default timeout value?

File: must-gather/collection-scripts/logs/gather_logs_backup (hunk 3 above)
Comment (kaovilai) [reply must-gather/collection-scripts/logs/gather_logs_backup:20]: in theory no.

Comment (kaovilai) [reply must-gather/collection-scripts/logs/gather_logs_backup:20]: From further testing and doc digging.. --request-timeout has very different effects to `timeout` in velero container.

--request-timeout is [The length of time to wait before giving up on a single server request.](https://kubernetes.io/docs/reference/kubectl/kubectl/)

whereas, if the server has responded, request-timeout do not work to kill a stuck velero CLI process.

So I propose we keep both for the `$timeout` defined case.

**Comment by kaovilai** [comment]
checking for ways to write other than empty file when timeout is triggered.

**Comment by kaovilai** [comment]
@shubham-pampattiwar when timeout occurs, the file (backup/restore-log/describe-<name>) will contain output like the following if timeout killed the program.
```
Defaulted container "velero" out of: velero, openshift-velero-plugin (init), velero-plugin-for-aws (init), velero-plugin-for-csi (init)
I0913 00:38:32.700413  121602 request.go:665] Waited for 1.188090857s due to client-side throttling, not priority and fairness, request: GET:https://172.30.0.1:443/apis/k8s.cni.cncf.io/v1?timeout=32s
command terminated with exit code 124

```

Specifically `124` is an exit code given by `timeout` if timeout killed the program. I believe the verbosity seen in the file came from oc exec reporting this exit code.

**Comment by shubham-pampattiwar** [comment]
@kaovilai LGTM!

**Review by shubham-pampattiwar — APPROVED** [review]

**Pull request merged**